scico/functional/_norm.py

```diff
@@ -477,3 +479,110 @@ def prox(
         svdU, svdS, svdV = snp.linalg.svd(v, full_matrices=False)
         svdS = snp.maximum(0, svdS - lam)
         return svdU @ snp.diag(svdS) @ svdV
+
+
+class TV2DNorm(Functional):
```

Comment:
It would be better for this to be able to handle arbitrary numbers of dimensions.

Reply:
@bwohlberg , we can extend it to 3D by adding another set of conditional statements if axis == 2. However, it doesn't look straightforward to extend it to arbitrary number of dimensions because we need to make in-place updates and in `Jax` we do this using `arr.at().set()` framework. But while numpy allows for indexing using `.take()` function, `Jax` doesn't seem to allow it. Do you know of a way to update immutable arrays in Jax by indexing using equivalent of `take()` to replace `at().set()`?

Reply:
Let's come back to this after having resolved the other issues.